src/value/owned.rs

```diff
@@ -335,3 +335,118 @@ impl Element for OwnedElement {
         }
     }
 }
+
+#[cfg(test)]
+mod value_tests {
+    use crate::result::IonResult;
+    use crate::types::decimal::Decimal;
+    use crate::types::timestamp::Timestamp;
+    use crate::value::owned::{OwnedElement, OwnedSequence, OwnedSymbolToken, OwnedValue};
+    use crate::value::{AnyInt, Element, IntAccess, SymbolToken};
+
+    fn extract_text<T: SymbolToken>(tok: &T) -> String {
+        tok.text().unwrap().into()
+    }
+
+    fn create_element_from_value(value: OwnedValue) -> OwnedElement {
+        let annotations = vec!["foo", "bar", "baz"];
+        let owned_elem =
+            OwnedElement::new(annotations.iter().map(|s| (*s).into()).collect(), value);
+        owned_elem
+    }
+
+    #[test]
+    fn test_as_any_int() -> IonResult<()> {
+        let owned_elem = create_element_from_value(OwnedValue::Integer(AnyInt::I64(100)));
+        let expected_elem = AnyInt::I64(100);
+
+        assert_eq!(&expected_elem.as_i64(), &owned_elem.as_i64());
+        assert_eq!(&expected_elem.as_big_int(), &owned_elem.as_big_int());
+        Ok(())
+    }
+
+    #[test]
+    fn test_as_float() -> IonResult<()> {
+        let owned_elem = create_element_from_value(OwnedValue::Float(305e1));
+        let expected_elem = Some(305e1);
+
+        assert_eq!(&expected_elem, &owned_elem.as_float());
+        Ok(())
+    }
```

Comment:
Might be worth considering (or adding an issue/TODO) to do the negative testing around the other `as` methods.  Another thing to consider (or add an issue/TODO)  is how to do test fixture style parameterization via something like `rstest`.

Specifically, as I look at each of these cases, they are really just testing the wrapping functionality and the ability to access the value through the `as` methods.  Creating a fixture for a similar equivalence class might be to just parameterize on `OwnedValue` and using a table or something to test that `as` methods are `None` except for the kind of value being wrapped.  You'll get the negative coverage this way and avoid writing a lot of boilerplate.

Reply:
Created an issue for this: #146